packages/react-shopify-app-bridge/src/Provider.tsx

```diff
@@ -220,51 +228,31 @@ export const Provider = ({
     });
   }, []);
 
-  let app = (
-    <GadgetUrqlProvider api={api}>
-      <InnerGadgetProvider
-        forceRedirect={forceRedirect}
-        isEmbedded={isEmbedded}
-        gadgetAppUrl={gadgetAppUrl}
-        api={api}
-        originalQueryParams={originalQueryParams}
-        isRootFrameRequest={isRootFrameRequest}
-        type={coalescedType}
-        isInstallRequest={!!isInstallRequest}
-      >
-        {children}
-      </InnerGadgetProvider>
-    </GadgetUrqlProvider>
-  );
-
-  const shouldMountAppBridge = host && coalescedType != AppType.Standalone && (!isInstallRequest || inDestinationContext);
-
   console.debug("[gadget-rsab] provider rendering", {
     host,
     coalescedType,
-    isInstallRequest,
     isReady,
-    isEmbedded,
-    isRootFrameRequest,
-    inDestinationContext,
-    shouldMountAppBridge,
   });
 
-  // app bridge provider seems to prevent urql from sending graphql requests when it cannot communicate using postMessage when not embedded so we must skip using the app bridge provider on the very first redirect from shopify
-  if (shouldMountAppBridge) {
-    app = (
-      <AppBridgeReact.Provider
-        config={{
-          apiKey: shopifyApiKey,
-          host,
-          forceRedirect,
-        }}
-        router={router}
-      >
-        {app}
-      </AppBridgeReact.Provider>
-    );
-  }
-
-  return app;
+  return (
+    <GadgetUrqlProvider api={api}>
+      {shopifyGlobalDefined && coalescedType === AppType.Embedded ? (
```

Comment:
yah I think if app type is Embedded and there is no shopify Global we should fail loudly

Reply:
should we actually throw in those cases? i think it might make sense to set something on the `error` property of the `useGadget` hook. if we throw consumers will have to use an `ErrorBoundary` to handle errors

Reply:
i think this is the kind of error that only the developer can fix -- it isn't a result of the user's actions where their wifi is disconnected or whatever, it's busted code, so I think full throwing is fine?

Reply:
what would the experience be right now if you tried to load this in the admin and for some reason the app bridge script tag wasn't there (I'm thinking if someone upgrades the gadget package but doesn't get the script tag)

As long as they can get the message about what they need to do that's all I'm worried about